Source/Core/VideoCommon/UberShaderVertex.cpp

```diff
@@ -176,8 +176,45 @@ ShaderCode GenVertexShader(APIType api_type, const ShaderHostConfig& host_config
                "\n");
 
   // Hardware Lighting
-  WriteVertexLighting(out, api_type, "pos.xyz", "_norm0", "rawcolor0", "rawcolor1", "o.colors_0",
-                      "o.colors_1");
+  out.WriteFmt("// xfmem.numColorChans controls the number of color channels available to TEV,\n"
+               "// but we still need to generate all channels here, as it can be used in texgen.\n"
+               "// Cel-damage is an example of this.\n"
+               "float4 vertex_color_0, vertex_color_1;\n"
+               "\n");
+  out.WriteFmt("// To use color 1, the vertex descriptor must have color 0 and 1.\n"
+               "// If color 1 is present but not color 0, it is used for lighting channel 0.\n"
+               "bool use_color_1 = ((components & {0}) == {0}); // VB_HAS_COL0 | VB_HAS_COL1\n",
+               VB_HAS_COL0 | VB_HAS_COL1);
```

Comment:
The code in this area previously wrote `static_cast<u32>(VB_HAS_COL0 | VB_HAS_COL1)`.  Based on https://github.com/dolphin-emu/dolphin/pull/4601#pullrequestreview-78152510 I don't think the cast is necessary, so I've removed it.  (This line specifically also had it duplicated, which I've replaced with `{0}`).